scripts/sct_analyze_lesion.py

```diff
@@ -485,47 +504,43 @@ def ifolder2tmp(self):
         os.chdir(self.tmp_dir)  # go to tmp directory
 
 
-def main(args=None):
-    if args is None:
-        args = sys.argv[1:]
-
-    # get parser
+def main():
     parser = get_parser()
-    arguments = parser.parse(args)
+    arguments = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
 
     # set param arguments ad inputted by user
-    fname_mask = arguments["-m"]
+    fname_mask = arguments.m
 
     # SC segmentation
-    if '-s' in arguments:
-        fname_sc = arguments["-s"]
+    if arguments.s is not None:
```

Comment:
@YangHee-Min do you need to add this line, given that is always a default value when we declare the parser fields?

Reply:
@jcohenadad If there is a default I believe it is not necessary. However, in the case for this argument as seen [(here)](https://github.com/neuropoly/spinalcordtoolbox/blob/ee58d326fcda797a947dff823d0e106542680e05/scripts/sct_analyze_lesion.py#L58-L63), there is no default. Therefore I believe this was necessary due to there being no default value. 

I will look into it and do some tests related to default values and whatnot just to see how the commands behave.

Reply:
> @jcohenadad If there is a default I believe it is not necessary. However, in the case for this argument as seen [(here)](https://github.com/neuropoly/spinalcordtoolbox/blob/ee58d326fcda797a947dff823d0e106542680e05/scripts/sct_analyze_lesion.py#L58-L63), there is no default. Therefore I believe this was necessary due to there being no default value.
> 
> I will look into it and do some tests related to default values and whatnot just to see how the commands behave.

for the example you gave, you can simply move [this declaration](https://github.com/neuropoly/spinalcordtoolbox/blob/ee58d326fcda797a947dff823d0e106542680e05/scripts/sct_analyze_lesion.py#L521) in the parser definition (as opposed to in the main)-- it will save you a lot of lines of code.

Reply:
Does something along the lines of this commit (Commit [fe41bbe](https://github.com/neuropoly/spinalcordtoolbox/commit/fe41bbeed8aafce11c83176d296af2bd857c3985)) do the trick?